### PR TITLE
Documentation tweak for sftp servers

### LINKED ISSUE #385: Unable to respond to SFTP REALPATH request - get "do_realpath: buffer error: incomplete message"
Sorry to finally have to open an issue for this, but I've been combing through source code and examples and everything I can find to try and just get it so that I can respond to a REALPATH request.

I've got authentication working correctly, and the basic setup seems to be working.

```js
        // ... 
        sftpStream.on('REALPATH', function (reqid,filename) {
          console.warn("REALPATH Request for filename: '"+filename+" has REQ ID: "+reqid);
            sftpStream.name(reqid, {filename: "/fribblefargz", longname: "/fribblefargz"});
            sftpStream.end(); //I've tried with and without this, still doesn't work...
        });
        // ...
```

Then if I try to connect to this server:

```sh
sftp -v  -P 8022 brady@localhost
```

As soon as I input my correct password, the REALPATH debug info will show (from above), and then sftp will exit with this error (including debug output from sftp):

```sh
debug1: Authentication succeeded (password).
Authenticated to localhost ([127.0.0.1]:8022).
debug1: channel 0: new [client-session]
debug1: Entering interactive session.
debug1: Sending environment.
debug1: Sending env LANG = en_US.UTF-8
debug1: Sending subsystem: sftp
Connected to localhost.
do_realpath: buffer error: incomplete message
```

Am I missing something? Am I invoking the `.name()` method the wrong way?

I also would love to help update the documentation in this area to help make this clearer for first-time users.

Thank you for making a module that looks like it will single-handedly save my project!!!

### PR DESCRIPTION
**Pull request by uberbrady**
Just added the SFTP server example from #383, mentioned by @mscdex 

Closes #385

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -541,6 +541,8 @@ new ssh2.Server({
 });
 ```
 
+* There are two more examples in the [examples](examples) directory. One is a chat
```
Comment (mscdex) [inline README.md]: It's probably best to tweak this to just say `There are more examples in the `examples` directory`, that way more examples can be added without modifying the readme.

**Comment by mscdex** [comment]
The filename of the new example should probably be a bit more specific like `sftp-server-download-only.js` or something to that effect.

**Comment by uberbrady** [comment]
Ok, I've made the tweaks you suggested.

If we missed any, let me know and I'll take another swing at it.

Thanks again!

**Comment by mscdex** [comment]
Landed in 87960f7dff2f2a92f7b9b3f79b4623d8d4062fdc. Thanks!

**Pull request closed without merging**